### DISCUSSION TITLE
Redelegation completion time

### DISCUSSION THREAD

**Discussion by boaz30333** [cosmos/community#15] [Chain Builders (Cosmos SDK)] [answered]
Hi all,
I'm testing cosmos redelegation, and I notice that there is a completion time for the redelegation transaction. (2 days in testnet)
Is the main benefit of redelegation over undelegate + delegate, that we are avoiding this cooldown period?(link for the transaction
https://explorer.theta-testnet.polypore.xyz/transactions/15FD6338E7C18AAEC9B99B6FC843D01989392FFE7D35DE972AF9E8405EE686B9)

Thanks in advance!

**Comment by alexanderbez** [comment]
What's the exact question? Redelegation is just moving your stake essentially. It can be seen as an unbonding + immediate bond (to a different validator).

**Comment by atheeshp** [comment, marked as answer]
Hey @boaz30333, addition to what @alexanderbez mentioned:
> Redelegation is just moving your stake essentially

Yes, the redelegation effects immediately to the new validator but if you have to redelegate again to another validator, you have to wait for unbonding period time.

> It can be seen as an unbonding + immediate bond (to a different validator).

may be we can say the redelegation works as like immediate unbond + bond (but we cannot escape the unbonding period to redelegate again) to other validator.
